app/models/availability.rb

```diff
@@ -39,4 +41,15 @@ def self.process_data
         { 'Date' => date, 'Count' => count }
       end
   end
+
+  private
+
+  def cancel_duplicates
+    previous_availabilities = person.availabilities.for_time_span(start_time..end_time).order(:created_at)
+    previous_availabilities.each do |a|
+      if start_time == a.start_time && end_time == a.end_time
+        a.update(status: "Cancelled") unless self == a
```

Comment:
Right. If that's the case, my suggestion is to replace this whole block with something like: 
```ruby
# cancel all availabilities that are exact duplicates of this one
person.availabilities.where(start_time: self.start_time, end_time: self.end_time)
  .where.not(id: self.id).update_all(status: "Cancelled")
```

I haven't tested this but I think that captures the essence of what this is trying to accomplish?